zingolib/src/wallet/data.rs

```diff
@@ -1738,8 +1738,11 @@ pub mod summaries {
         /// Output is not spent.
         Unspent,
         /// Output is pending spent.
-        /// The transaction consuming this output is pending.
-        PendingSpent(TxId),
+        /// The transaction consuming this output has been transmitted.
+        TransmittedSpent(TxId),
+        /// Output is pending spent.
+        /// The transaction consuming this output has been detected in the mempool.
+        MempoolSpent(TxId),
```

Comment:
I believe this is @zancas 's wet code. If variants of SpendSummary correspond to to variants of ConfirmationStatus, use an enum wrapper!

Reply:
We have a method that creates one from an  `Option<(Txid, ConfirmationStatus)>`. We could have it be defined as a newtype wrapper (or enum wrapper, like `enum SpendSummary { Unspent, SpendStatus(ConfirmationStatus)}`, I do think it's a tradeoff. There's a readability benefit to having the definition show all the possible variants at the top level of the enum, especially where they're all have their own variant documentation.

Reply:
This is marked outdated by github.